gobblin-compaction/src/main/java/gobblin/compaction/mapreduce/avro/AvroKeyReducer.java

```diff
@@ -0,0 +1,54 @@
+/* (c) 2015 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package gobblin.compaction.mapreduce.avro;
+
+import java.io.IOException;
+
+import org.apache.avro.generic.GenericRecord;
+import org.apache.avro.mapred.AvroKey;
+import org.apache.avro.mapred.AvroValue;
+import org.apache.hadoop.io.NullWritable;
+import org.apache.hadoop.mapreduce.Reducer;
+
+
+/**
+ * Reducer class for compaction MR job for Avro data.
+ *
+ * @author ziliu
+ */
+public class AvroKeyReducer extends
+    Reducer<AvroKey<GenericRecord>, AvroValue<GenericRecord>, AvroKey<GenericRecord>, NullWritable> {
+  private AvroKey<GenericRecord> outKey;
+
+  @Override
+  protected void setup(Context context) throws IOException, InterruptedException {
+    outKey = new AvroKey<GenericRecord>();
+  }
+
+  @Override
+  protected void reduce(AvroKey<GenericRecord> key, Iterable<AvroValue<GenericRecord>> values, Context context)
```

Comment:
As far as I understand, this reduce method takes the last value with the same key. Values with the same key can come in any order, so this reducer is essentially randomly picking one record out of all records with the same index in the input. If they are all identical, that is ok. But for a snapshot-delta compaction, for example, this wouldn't work. This should be clear in the javadoc.